### ISSUE TITLE
Fix test `yield_calls_park_before_scheduling_again`

### ISSUE THREAD

**Issue by Darksonn** [tokio-rs/tokio#6536]
If you run the following command from the tokio subdirectory:
```
RUSTFLAGS="-Dwarnings --cfg loom  -C debug_assertions" LOOM_MAX_PREEMPTIONS=1 LOOM_MAX_BRANCHES=10000 cargo test --lib --release --features full -- --test-threads=1 --nocapture loom_multi_thread::yield_now::yield_calls_park_before_scheduling_again
```
then this test will fail:

https://github.com/tokio-rs/tokio/blob/f6eb1ee19687bfd6cc7bf5c675d946970854013e/tokio/src/runtime/tests/loom_multi_thread/yield_now.rs#L5-L30

then the `assert_eq!(park_count + 1, new_park_count)` assertion will fail like this:
```
running 1 test
test runtime::tests::loom_multi_thread::yield_now::yield_calls_park_before_scheduling_again ...
thread 'runtime::tests::loom_multi_thread::yield_now::yield_calls_park_before_scheduling_again' panicked at tokio/src/runtime/tests/loom_multi_thread/yield_now.rs:22:17:
assertion `left == right` failed
  left: 1
 right: 0
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
// and some more stuff that's not important
```
This test should be fixed.

---

Details on why the test fails:

The `yield_now` function uses [a special defer queue](https://github.com/tokio-rs/tokio/blob/f6eb1ee19687bfd6cc7bf5c675d946970854013e/tokio/src/runtime/scheduler/defer.rs) to make sure that the task that is using `yield_now` doesn't get polled again until we have checked the IO/timer drivers for new IO or timer events. This ensures that IO and timer events are given higher priority than the `yield_now` task. Behind the scenes, the IO and timer drivers are checked by calling the `park_timeout` function with a timeout of zero. The test in question verifies that the IO/timer drivers are actually checked before the task gets scheduled again.

The test is able to detect whether parking has happened using this counter:

https://github.com/tokio-rs/tokio/blob/f6eb1ee19687bfd6cc7bf5c675d946970854013e/tokio/src/runtime/park.rs#L35-L39

However, there is one situation where the runtime attempts to park, but is unable to actually do so:

https://github.com/tokio-rs/tokio/blob/f6eb1ee19687bfd6cc7bf5c675d946970854013e/tokio/src/runtime/scheduler/multi_thread/park.rs#L70-L77

Basically, the call to `try_lock` fails if a different worker thread is also calling `park_timeout` at the same time. In that case, the current thread doesn't actually perform the park operation, since there is another thread that will take care of IO/timer events for us.

To fix this we, need to increment the `CURRENT_THREAD_PARK_COUNT` counter even if `try_lock` fails.

**Comment by evanrittenhouse** [comment]
Feel free to assign this to me, I can pick it up
